File: mobile/src/components/EmptyState.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { colors, spacing } from '../theme/colors';

export default function EmptyState({ title = 'لا توجد بيانات', subtitle, icon = '📭' }) {
  return (
    <View style={styles.container}>
      <Text style={styles.icon}>{icon}</Text>
      <Text style={styles.title}>{title}</Text>
      {subtitle && <Text style={styles.subtitle}>{subtitle}</Text>}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: spacing.xxxl,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: { fontSize: 40, marginBottom: spacing.md },
  title: { fontSize: 14, fontWeight: '600', color: colors.text, textAlign: 'center' },
  subtitle: { fontSize: 12, color: colors.textMuted, marginTop: 4, textAlign: 'center' },
});

File: mobile_flutter/lib/widgets/empty_state.dart
import 'package:flutter/material.dart';
import '../theme/colors.dart';

class EmptyState extends StatelessWidget {
  final String title;
  final String? subtitle;
  final String icon;

  const EmptyState({
    super.key,
    this.title = 'لا توجد بيانات',
    this.subtitle,
    this.icon = '📭',
  });

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(AppSpacing.xxxl),
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text(icon, style: const TextStyle(fontSize: 40)),
          const SizedBox(height: AppSpacing.md),
          Text(title,
              textAlign: TextAlign.center,
              style: const TextStyle(fontSize: 14, fontWeight: FontWeight.w600, color: AppColors.text)),
          if (subtitle != null) ...[
            const SizedBox(height: 4),
            Text(subtitle!,
                textAlign: TextAlign.center,
                style: const TextStyle(fontSize: 12, color: AppColors.textMuted)),
          ],
        ],
      ),
    );
  }
}
